cp23/hpi-fun2/cubes/executables/generator.cpp

```diff
@@ -0,0 +1,159 @@
+#include <testlib.h>
+#include <bits/stdc++.h>
+
+#include <FastNoiseLite.h>
+
+using namespace std;
+
+const long long MAX_COORDINATE = 1e12;
+const int MAX_N = 1e5;
+
+const string_view SAMPLE1 = R"(7
+2 2 2
+2 2 1
+2 2 3
+2 1 2
+2 3 2
+1 2 2
+3 2 2
+)";
+
+const string_view SAMPLE2 = R"(2
+1 1 1
+1 1 3
+)";
+
+template <class F>
+void testcase(string name, string desc, F f)
+{
+    ofstream desc_file(name + ".desc");
+    desc_file << desc;
+    string in_file = name + ".in";
+    freopen(in_file.c_str(), "w", stdout);
+    f();
+}
+
+void predefined(string name, string desc, string_view content)
+{
+    testcase(name, desc, [&]()
+             { cout << content; });
+}
+
+void sample(int num, string_view content)
+{
+    auto num_str = toString(num);
+    predefined("sample" + num_str, "Sample #" + num_str, content);
+}
+
+vector<array<long long, 3>> generate_coords(int width, int length, int height)
+{
+    vector<array<long long, 3>> coords;
+
+    const long long base_x_coordinate = rnd.next(MAX_COORDINATE);
+    const long long base_y_coordinate = rnd.next(MAX_COORDINATE);
+    const long long base_z_coordinate = rnd.next(MAX_COORDINATE);
+
+    // Threshold for the cloud block
+    const float THRESHOLD = 0.4;
+
+    FastNoiseLite noiseGenerator;
+
+    // Configure the noise
+    noiseGenerator.SetNoiseType(FastNoiseLite::NoiseType_ValueCubic);
+    noiseGenerator.SetFrequency(0.1);
+
+    for (int x = 0; x < width; x++)
+    {
+        for (int y = 0; y < length; y++)
+        {
+            for (int z = 0; z < height; z++)
+            {
+                // Generate noise value
+                float noise = noiseGenerator.GetNoise((float)x, (float)y, (float)z);
+
+                // If noise value is above threshold, generate a cloud block
+                if (noise > THRESHOLD)
+                    coords.push_back({base_x_coordinate + x, base_y_coordinate + y, base_z_coordinate + z});
```

Comment:
Also note that your coords will always appear in a sorted order. You might want to use testlibs `shuffle` at the end of this method.